zarnitsa/DataAugmenterInternally.py

```diff
@@ -67,35 +53,61 @@ def _prepare_data_to_aug(
             return data, data.sample(0)
 
     def augment_column_permut(
-        self, col: pd.Series, freq=0.2, return_only_aug=False
+        self, col: pd.Series, n_to_aug=None, freq=0.2, return_only_aug=False
     ) -> pd.Series:
-        """Augment column data using permutations. Pandas column"""
+        """
+        Augment column data using permutations. Pandas column
+        param: col: pandas series object
+        param: n_to_aug: defined number of augmented examples
+        param: freq: part of the data which will be the base for augmentation
+        param: return_only_aug: ask return only augmented data
+        """
         not_to_aug, to_aug = self._prepare_data_to_aug(col, freq=freq)
         indices_to_permute = to_aug.index
         to_aug = to_aug.sample(frac=1.0)
         to_aug.index = indices_to_permute
+        if n_to_aug:
+            n_to_aug = n_to_aug if n_to_aug < to_aug.shape[0] else to_aug.shape[0]
+            to_aug = to_aug.sample(n_to_aug)
         return to_aug if return_only_aug else pd.concat([not_to_aug, to_aug])
 
     def augment_column_norm(
-        self, col: pd.Series, freq=0.2, return_only_aug=False
+        self, col: pd.Series, n_to_aug=None, freq=0.2, return_only_aug=False
     ) -> pd.Series:
-        """Augment column data using normal distribution. Pandas column"""
+        """
+        Augment column data using normal distribution. Pandas column
+        param: col: pandas series object
+        param: n_to_aug: defined number of augmented examples
+        param: freq: part of the data which will be the base for augmentation
+        param: return_only_aug: ask return only augmented data
+        """
         not_to_aug, to_aug = self._prepare_data_to_aug(col, freq=freq)
         column_std = col.std()
-        to_aug = to_aug.apply(
-            lambda value: np.random.normal(value, column_std)
-        )
+        to_aug = to_aug.apply(lambda value: np.random.normal(value, column_std))
+        if n_to_aug:
+            n_to_aug = n_to_aug if n_to_aug < to_aug.shape[0] else to_aug.shape[0]
+            to_aug = to_aug.sample(n_to_aug)
         return to_aug if return_only_aug else pd.concat([not_to_aug, to_aug])
 
     def augment_column_uniform(
-        self, col: pd.Series, freq=0.2, n_sigm=3, return_only_aug=False
+        self, col: pd.Series, n_to_aug=None, freq=0.2, n_sigm=3, return_only_aug=False
```

Comment:
I think that it is usually better to add new args at the end of args list if it is possible (in backward compatibility purpose), but I think we shouldn't care about backward compatibility at the moment and it's ok :)

Reply:
heh) well, i just thought that its better to place argument ordered by their importance